presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,549 @@ Tuning Presto
+
+
+
+.. _tuning-pref-session:
+
+Session properties
+------------------
+
+``processing_optimization``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (``disabled``, ``columnar`` or ``columnar_dictionary``)
+ * **Default value:** ``optimizer.processing-optimization`` (``false``)
+ * **Description:** See :ref:`optimizer.processing-optimization<tuning-pref-optimizer>`.
+
+
+``execution_policy``
+^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (``all-at-once`` or ``phased``)
+ * **Default value:** ``query.execution-policy`` (``all-at-once``)
+ * **Description:** See :ref:`query.execution-policy <tuning-pref-query>`.
+
+
+``hash_partition_count``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer``
+ * **Default value:** ``query.initial-hash-partitions`` (``100``)
+ * **Description:** See :ref:`query.initial-hash-partitions <tuning-pref-query>`.
+
+
+``optimize_hash_generation``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``optimizer.optimize-hash-generation`` (``true``)
+ * **Description:** See :ref:`optimizer.optimize-hash-generation <tuning-pref-optimizer>`.
+
+
+``orc_max_buffer_size``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``hive.orc.max-buffer-size`` (``8 MB``)
+ * **Description:** See :ref:`hive.orc.max-buffer-size <tuning-pref-hive>`.
+
+
+``orc_max_merge_distance``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``hive.orc.max-merge-distance`` (``1 MB``)
+ * **Description:** See :ref:`hive.orc.max-merge-distance <tuning-pref-hive>`.
+
+
+``orc_stream_buffer_size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``hive.orc.max-buffer-size`` (``8 MB``)
+ * **Description:** See :ref:`hive.orc.max-buffer-size <tuning-pref-hive>`.
+
+
+``plan_with_table_node_partitioning``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``true``
+ * **Description:** **Experimental.** Adapt plan to use backend partitioning. By setting this property you allow to use partitioning provided by table layout itself while collecting required data. This may allow to utilize optimization of table layout provided by specific connector. In particular, when this is set presto will try to partition data for workers in a way that each workers gets a chunk of data that comes from one backend partition. It can be particularly useful due to the I/O distribution optimization in table partitioning. Note that this property may only be utilized if given projection uses all columns used for table partitioning inside connector.
```

Comment:
This simply means that when planning a query Presto will take advantage of pre-organized data.  This property will likely be remove soon and was only added as a way to disable the feature if there was a problem.